### PR TITLE
Adding new backward-compatible handlers 

### PR DESCRIPTION
**Pull request by acostapazo**

### CODE REVIEW

**Comment by acostapazo** [comment]
As we're installing locally meiga, we're overwriting lume dependency and the new method signature is not compatible!

File: README.md [hunk 1]
```diff
@@ -326,48 +330,51 @@ If you need to add some arguments as a parameters, use **success_args** and **fa
...
+    assert param_1 == 1
+    assert isinstance(result, Result)
+    assert result.value == Error()
+    assert param_2 == 2
+
+
+@meiga
+def run():
+    result.handle(
+        n_success_handler=OnSuccessHandler(func=success_handler, args=args),
```
Comment (franciscorode) [inline README.md:372]: typo: expects the `o` in parameter name `n_success_handler`

File: meiga/result.py [hunk 2]
```diff
@@ -107,56 +111,55 @@ def unwrap_or_throw(self) -> TS:
 
     def unwrap_or_else(
         self,
-        on_failure: Callable[..., Any],
-        failure_args: Optional[Any] = None,
+        on_failure_handler: OnFailureHandler = None,  # Default has to be None to be compatible with deprecated signature
```
Comment (franciscorode) [inline meiga/result.py:114]: If can be `None` I think it should be typed as `Optional`

File: meiga/result.py [hunk 3]
```diff
@@ -107,56 +111,55 @@ def unwrap_or_throw(self) -> TS:
...
             return self.value
         return None
 
     def handle(
         self,
-        on_success: Optional[Callable[..., None]] = None,
-        on_failure: Optional[Callable[..., None]] = None,
-        success_args: Optional[Any] = None,
-        failure_args: Optional[Any] = None,
+        on_success_handler: OnSuccessHandler = None,
```
Comment (franciscorode) [inline meiga/result.py:145]: Idem

File: meiga/handlers.py [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+from meiga.misc import get_args_list
+
+
+class Handler:
+    def __init__(self, func: Callable[..., None], args: Iterable = None):
+        self.func = func
+        self.args = args
+
+    def execute(self, result) -> None:
```
Comment (franciscorode) [inline meiga/handlers.py:11]: Here can we type the result parameter as `Result`?

**Review by franciscorode — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (acostapazo) [reply README.md]: Alright!

File: meiga/result.py (hunk 3 above)
Comment (acostapazo) [reply meiga/result.py:145]: Idem

File: meiga/handlers.py (hunk 4 above)
Comment (acostapazo) [reply meiga/handlers.py:11]: Yes, we should, but it is a circular dependency (handler is called by Result). I tried using quotes but didn't work it!

**Review by franciscorode — APPROVED** [review]

**Pull request merged**